frostsnap_core/src/message.rs

```diff
@@ -77,7 +78,7 @@ pub enum CoordinatorToStorageMessage {
 #[derive(Clone, Debug, bincode::Encode, bincode::Decode)]
 pub struct DeviceToCoordindatorMessage {
     pub from: DeviceId,
-    pub body: DeviceToCoordinatorBody,
+    pub body: Box<DeviceToCoordinatorBody>,
```

Comment:
Ok so the reason this helps is that DeviceToCoordinatorBody is a huge size because `KeyGenResponse` is huge which makes `DeviceSend` huge. I don't think this solves the root problem though which is that `KeyGenResponse` is huge.  `KeyGenResponse` is still making `DeviceToCoordinatorBody::SignatureShare` much bigger than it needs to be. The more targeted solution is to `Box` the array:

```rust
#[derive(Clone, Debug, bincode::Encode, bincode::Decode, Eq, PartialEq)]
pub struct KeyGenResponse {
    pub encrypted_shares: KeyGenProvideShares,
    pub nonces: Box<[Nonce; NONCE_BATCH_SIZE]>,
}
```

I wonder if we can come up with a more general way to stop this from happening again. It seems like a bug you could easily add in a release and wait for end users to notice it. Did clippy warn you of this? Maybe we can configure that check to be even more sensitive.

Reply:
Yep this was noticed from a clippy warning, we could make the clippy test fail on warnings, or something specific with [enum-variant-size-threshold](https://doc.rust-lang.org/nightly/clippy/lint_configuration.html?highlight=enum#enum-variant-size-threshold)